.appveyor.yml

```diff
@@ -7,10 +7,18 @@ environment:
   PYTHON: C:\Python38
   SCONS_CACHE_ROOT: "%HOME%\\scons_cache"
   SCONS_CACHE_LIMIT: 1024
+  OPTIONS: "debug_symbols=no verbose=yes progress=no builtin_libpng=yes"
+  EXTRA_ARGS: "warnings=all werror=yes"
   matrix:
-    - GD_PLATFORM: windows
-      TOOLS: yes
-      TARGET: release_debug
+  - GD_PLATFORM: windows
+    TARGET: release_debug
+    TOOLS: yes
+  - GD_PLATFORM: windows
+    TARGET: release
+    TOOLS: no
```

Comment:
There are two problems with additional an additional AppVeyor build:
- We only get one concurrent builds on AppVeyor, so all commits will take twice as long to build. This will mean slowdown for merging PRs that could otherwise be merged within minutes.
- AFAIK AppVeyor only provides 1 GB of cache, which we already fill with a single build, so having two builds will probably mean invalidating the cache all the time and will defeat the purpose of having a cache. So builds will no longer take ~3-5 min but 40-50 min.

Reply:
> We only get one concurrent builds on AppVeyor, so all commits will take twice as long to build.

My personal experience is that it's the Travis CI tests that are currently taking the time, but I think it depends on the changes.
From my last PR (36079):
[Appveyor](https://ci.appveyor.com/project/akien-mga/godot/builds/30694749): 3 min 13 sec
[Travis](https://travis-ci.org/godotengine/godot/builds/648436097?utm_source=github_status&utm_medium=notification): 11 min 45 sec 

> AFAIK AppVeyor only provides 1 GB of cache,

This is true. From Appveyor's Caching data between builds [page](https://www.appveyor.com/docs/build-cache/):
> Total cache size per account consist of all caches from all projects. If projects contain more than one job in build matrix or in parallel testing, each job has its own separate cache, which cannot be shared with other jobs or projects.
 <div><table>
<tr>
<td>Free</td>
<td>Basic</td>
<td>Pro</td>
<td>Premium</td>
</tr><tr>
<td>1 GB</td>
<td>1 GB</td>
<td>5 GB</td>
<td>20 GB</td>
</tr>
</table></div>

which we already fill with a single build

As far as I can tell this is not the case. From my last build:
> [00:00:48] Cache 'C:\Users\appveyor\scons_cache' - Downloading (226,177,031 bytes)

So we're currently only using around a quarter of the allocated cache.

Reply:
OK, well I guess we can give it a try and see how it goes under usual load.

I'm about to merge `vulkan` into `master`, which disables some CI tests in the process, so we might want to wait a bit for warnings to be fixed before rebasing and merging this.

Reply:
> we might want to wait a bit for warnings to be fixed before rebasing and merging this.

We should merge #33670, #33731 and #36097 to resolve all the existing warnings first anyway.